exporter/exporterparser/zipkin.go

```diff
@@ -106,89 +110,250 @@ func ZipkinExportersFromYAML(config []byte) (tes []exporter.TraceExporter, doneF
 	return
 }
 
-func newZipkinExporter(finalEndpointURI, defaultServiceName, defaultLocalEndpointURI string) (*zipkinExporter, error) {
-	localEndpoint, err := openzipkin.NewEndpoint(defaultServiceName, defaultLocalEndpointURI)
-	if err != nil {
-		return nil, err
+func newZipkinExporter(finalEndpointURI, defaultServiceName, defaultLocalEndpointURI string, uploadPeriod time.Duration) (*zipkinExporter, error) {
+	var opts []zipkinhttp.ReporterOption
+	if uploadPeriod > 0 {
+		opts = append(opts, zipkinhttp.BatchInterval(uploadPeriod))
 	}
-	reporter := http.NewReporter(finalEndpointURI)
-	defaultExporter := zipkin.NewExporter(reporter, localEndpoint)
+	reporter := zipkinhttp.NewReporter(finalEndpointURI, opts...)
 	zle := &zipkinExporter{
 		endpointURI:             finalEndpointURI,
-		defaultExporter:         defaultExporter,
 		defaultServiceName:      defaultServiceName,
 		defaultLocalEndpointURI: defaultLocalEndpointURI,
-		serviceNameToExporter:   make(map[string]*zipkin.Exporter),
+		reporter:                reporter,
 	}
+	return zle, nil
+}
 
-	// Ensure that we add the default reporter's Close functions
-	zle.shutdownFns = append(zle.shutdownFns, reporter.Close)
+func zipkinEndpointFromNode(node *commonpb.Node, serviceName string, prefixFunc func(string) string) (*zipkinmodel.Endpoint, error) {
```

Comment:
nit: seeing how the prefixFunc is used perhaps it will be better to use an array of literal strings to avoid allocations. 

Personally I would have 3 functions: one with a bool addRemotePrefix and two others like zipkinLocalEndpointFromNode and zipkinRemoteEndpointFromNode that just call the first with the proper value for addRemotePrefix. This way the allocations can be avoided and the prefix function is not needed.